WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormDataModel.swift

```diff
@@ -0,0 +1,81 @@
+import Yosemite
+
+/// Describes a data model that contains necessary properties for rendering a product form (`ProductFormViewController`).
+protocol ProductFormDataModel {
+    // General
+    var siteID: Int64 { get }
+    var productID: Int64 { get }
+    var name: String { get }
+    var description: String? { get }
+    var shortDescription: String? { get }
+
+    // Settings
+    var virtual: Bool { get }
+    var downloadable: Bool { get }
+    var permalink: String { get }
+
+    // Images
+    var images: [ProductImage] { get }
+
+    // Price
+    var regularPrice: String? { get }
+    var salePrice: String? { get }
+    var dateOnSaleStart: Date? { get }
+    var dateOnSaleEnd: Date? { get }
+    var taxStatusKey: String { get }
+    var taxClass: String? { get }
+
+    // Shipping
+    var weight: String? { get }
+    var dimensions: ProductDimensions { get }
+    var shippingClass: String? { get }
+    var shippingClassID: Int64 { get }
+
+    // Inventory
+    var sku: String? { get }
+    var manageStock: Bool { get }
+    var stockStatus: ProductStockStatus { get }
+    var stockQuantity: Int64? { get }
+    var backordersKey: String { get }
+}
+
+// MARK: Helpers that can be derived from `ProductFormDataModel`
+//
+extension ProductFormDataModel {
+    /// Returns the full description without the HTML tags and leading/trailing white spaces and new lines.
+    var trimmedFullDescription: String? {
+        guard let description = description else {
+            return nil
+        }
+        return description.removedHTMLTags.trimmingCharacters(in: .whitespacesAndNewlines)
+    }
+
+    /// Whether shipping settings are available for the product.
+    var isShippingEnabled: Bool {
+        return downloadable == false && virtual == false
+    }
+
+    /// Returns `ProductTaxStatus` given the raw value from `taxStatusKey` field.
+    var productTaxStatus: ProductTaxStatus {
+        return ProductTaxStatus(rawValue: taxStatusKey)
+    }
+
+    /// Returns `ProductBackordersSetting` given the raw value from `backordersKey` field.
+    var backordersSetting: ProductBackordersSetting {
+        return ProductBackordersSetting(rawValue: backordersKey)
+    }
```

Comment:
Q: is it worth to add unit tests to these functions?

Reply:
yes! we have some existing unit tests and I just added some:

- `trimmedFullDescription`: [`Product_ProductFormTests.testTrimmedFullDescriptionWithLeadingNewLinesAndHTMLTags`](https://github.com/woocommerce/woocommerce-ios/blob/09573496fe95d51dbeb4125a8fb57ea2e76cfd2c/WooCommerce/WooCommerceTests/ViewRelated/Products/Product+ProductFormTests.swift#L10-L15)
- `isShippingEnabled`: [`Product_ShippingTests`](https://github.com/woocommerce/woocommerce-ios/blob/fcc5b976b691127efbbd215a5cef99577ee28242/WooCommerce/WooCommerceTests/Extensions/Product%2BShippingTests.swift)
- `productTaxStatus` and `backordersSetting`: just added in https://github.com/woocommerce/woocommerce-ios/pull/2552/commits/27ac6231e47020fa94dd7d95f22750cd757582bd